### PR TITLE
Feature/system page

### PR DESCRIPTION
**Pull request by mstojanovicbloxico**
## Description

Overview tab for system page finished

## Type of change

- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update
- [ ] This change requires a configuration update

## How Has This Been Tested?

- [x I have performed a self-review of this code and spelling
- [ ] I have added comments in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Additional information

Include any additional information about these changes here.
Screenshot from 2021-11-24 08-01-33

### CODE REVIEW

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 1]
```diff
@@ -28,95 +33,224 @@ const RepositoryDetails = () => {
...
+  const pageLayout = usePageLayout({
+    header: <Header />,
     footer: <PageLayout.Footer />,
   })
 
+  const renderOverview = () => (
+    <Overview>
+      <ProgressBarWrapper>
+        <ProgressBar
+          data={AllTestKinds}
```
Comment (toljaEmbroker) [inline frontend/apps/web/src/pages/system/_name.tsx:73]: These are all mocks, we need real data from model to be imported and used.

Comment (toljaEmbroker) [reply frontend/apps/web/src/pages/system/_name.tsx:73]: also, upgrade the Progress bar (but create an issue for that and do it in a separate PR so that Voja can use it as well) so that onClick is optional.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 2]
```diff
@@ -28,95 +33,224 @@ const RepositoryDetails = () => {
...
-                  Passing tests <div />
-                </div>
-                <div>
-                  Failing tests <div />
-                </div>
-              </Legend>
-            </StackLayout>
-          </BoxLayout>
-        </DetailedView>
+        {activeTab === 1 ? renderOverview() : renderDetailedView()}
```
Comment (toljaEmbroker) [inline frontend/apps/web/src/pages/system/_name.tsx:253]: Create a component `<DetailedView {...here add props needed (system, testKinds, etc...}  />` and `<Overview/>`
You can add component in path:
`frontend/apps/web/src/components/system/DetailedView.tsx`

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 3]
```diff
@@ -28,95 +33,224 @@ const RepositoryDetails = () => {
...
+      </ProgressBarWrapper>
+      <ProgressBarWrapper>
+        <ProgressBar
+          data={AllTestStatus}
+          onClick={() => null}
+          legend
+        ></ProgressBar>
+      </ProgressBarWrapper>
+      <TableWrapper>
+        <Text type={'subtitle l'}>Subsystems (4)</Text>
```
Comment (toljaEmbroker) [inline frontend/apps/web/src/pages/system/_name.tsx:86]: count number of subsystems from model.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 4]
```diff
@@ -28,95 +33,224 @@ const RepositoryDetails = () => {
...
+                    size="small"
+                    color="warning"
+                  >
+                    Mediocre
+                  </Button>
+                )
+              },
+            },
+          }}
+          data={Subsystems}
```
Comment (toljaEmbroker) [inline frontend/apps/web/src/pages/system/_name.tsx:167]: use data from model as well.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 5]
```diff
@@ -28,95 +33,224 @@ const RepositoryDetails = () => {
...
-    header: (
+  const Header = () => {
+    return (
       <PageLayout.Header>
         <StackLayout>
           <Text type="heading 5">Systems</Text>
         </StackLayout>
+        <PageLayout.Tabs>
+          <PageLayout.Tab
+            onClick={() => setActiveTab(1)}
```
Comment (toljaEmbroker) [inline frontend/apps/web/src/pages/system/_name.tsx:47]: As we talked about, use setActiveTab('overview')

and also add something like this (now we use navigate instead of history, so check how it is now):  
`useEffect(()=>{history.push({search:'overview')},[activeTab])`

**Review by toljaEmbroker — CHANGES_REQUESTED** [review]

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 1 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx]: Data from model used

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 2 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx]: Done

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 3 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx]: Done

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 4 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx]: Done

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 5 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx]: Done

File: frontend/apps/web/src/components/system/DetailedView.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  MatrixMap,
+  StackLayout,
+  Text,
+} from '@filecoin/ui'
+import ReactTooltip from 'react-tooltip'
+import styled from 'styled-components'
+
+interface Props {
+  model: Model
+  modelName: string
```
Comment (aleksaToljic) [inline frontend/apps/web/src/components/system/DetailedView.tsx:17]: let's call this system ili systemName.

File: frontend/apps/web/src/components/system/Overview.tsx [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    <Wrapper>
+      <ProgressBarWrapper>
+        <ProgressBar
+          data={system.testKindStats.percentages.map(
+            ({ kind, percentage }) => ({
+              name: kind,
+              percentage,
+            }),
+          )}
+          onClick={() => null}
```
Comment (aleksaToljic) [inline frontend/apps/web/src/components/system/Overview.tsx:40]: This is now optional, so we can remove it.

File: frontend/apps/web/src/components/system/Overview.tsx [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+        ></ProgressBar>
+      </ProgressBarWrapper>
+      <TableWrapper>
+        <Text type={'subtitle l'}>Subsystems ({totalSubsystems})</Text>
+        <TableStyled
+          variant="default"
+          columns={{
+            Subsystems: {
+              header: 'Subsystem',
+              width: 150,
```
Comment (aleksaToljic) [inline frontend/apps/web/src/components/system/Overview.tsx:63]: To avoid 100% static tables (fixedWidth), let's leave one column width responsive, so what you thing will need the most space, remove the width prop on that column.

File: frontend/apps/web/src/components/system/Overview.tsx [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+              Cell: ({ data }) => <Text type="text s">{data.name}</Text>,
+            },
+            testKinds: {
+              header: 'Test Kinds',
+              width: 225,
+              Cell: ({ data }) => (
+                <ProgressBar
+                  onClick={() => navigate('/repository-details')}
```
Comment (aleksaToljic) [inline frontend/apps/web/src/components/system/Overview.tsx:71]: remove all redirecting here, since we don't have a link where to go

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 10]
```diff
@@ -1,65 +1,64 @@
...
-    return (
-      <PageLayout.Header>
-        <Text type="heading 5">Systems</Text>
+  const [activeTab, setActiveTab] = useState(
+    new URLSearchParams(window.location.search).get('tab') || 'overview',
+  )
 
+  useEffect(() => {
+    navigate({
+      pathname: '/system/Blockchain/',
```
Comment (aleksaToljic) [inline frontend/apps/web/src/pages/system/_name.tsx:33]: Try without the pathname, it should then just change the search for current link. If not than we need to check the current pathname, and add it here, we can't have it static like this.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 11]
```diff
@@ -69,139 +68,14 @@ const RepositoryDetails = () => {
...
-                  Failing tests <div />
-                </div>
-              </Legend>
-            </StackLayout>
-          </BoxLayout>
-        </DetailedView>
+        {activeTab === 'overview' ? (
+          <Overview model={model} modelName={params.name} />
+        ) : (
+          <DetailedView model={model} modelName={params.name} />
```
Comment (aleksaToljic) [inline frontend/apps/web/src/pages/system/_name.tsx:74]: hm I guess that even if you write overview2 as queryParam, it will open DetailedView, I'm not sure that we won't that. Let's add in our useEffect something like this.

```
const tabs = ['overview', 'detailedView']
if(!tabs.includes(activeTab)) {
  setActiveTab('overview')
}
```

**Review by aleksaToljic — CHANGES_REQUESTED** [review]

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 11 above)
Comment (aleksaToljic) [reply frontend/apps/web/src/pages/system/_name.tsx:74]: overview should be a default tab, so if queryParam is missing or invalid, we need to show overview, and update the link

File: frontend/apps/web/src/components/system/DetailedView.tsx (hunk 6 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/components/system/DetailedView.tsx:17]: done

File: frontend/apps/web/src/components/system/Overview.tsx (hunk 7 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/components/system/Overview.tsx:40]: done

File: frontend/apps/web/src/components/system/Overview.tsx (hunk 9 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/components/system/Overview.tsx:71]: done

File: frontend/apps/web/src/components/system/Overview.tsx (hunk 8 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/components/system/Overview.tsx:63]: Done. I removed width from both test columns

**Comment by aleksaToljic** [comment]
Picture in the description has issues with progressbar width, some are smaller some are bigger, is that fixed?

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 10 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx:33]: You are right. Done without pathname.

File: frontend/apps/web/src/pages/system/_name.tsx (hunk 11 above)
Comment (mstojanovicbloxico) [reply frontend/apps/web/src/pages/system/_name.tsx:74]: Done

**Comment by mstojanovicbloxico** [comment]
> Picture in the description has issues with progressbar width, some are smaller some are bigger, is that fixed?

That seems to be issue with either progressBar or Table. It happends on other pages too. I can look into it.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 12]
```diff
@@ -1,65 +1,68 @@
...
-        <Text type="heading 5">Systems</Text>
+  const [activeTab, setActiveTab] = useState<'overview' | 'detailedView'>(
+    'overview',
+  )
 
+  useEffect(() => {
+    if (!TABS.includes(activeTab)) {
+      setActiveTab('overview')
+    }
+    navigate({
```
Comment (aleksaToljic) [inline frontend/apps/web/src/pages/system/_name.tsx:37]: we can add navigate to else {} because we don't won't to navigate if the queryParam is wrong.

File: frontend/apps/web/src/pages/system/_name.tsx [hunk 13]
```diff
@@ -69,139 +72,14 @@ const RepositoryDetails = () => {
...
-                <div>
-                  Failing tests <div />
-                </div>
-              </Legend>
-            </StackLayout>
-          </BoxLayout>
-        </DetailedView>
+        {activeTab === 'overview' ? (
+          <Overview model={model} systemName={params.name} />
+        ) : (
```
Comment (aleksaToljic) [inline frontend/apps/web/src/pages/system/_name.tsx:77]: This can potentially be a glich. For example if it 'overview2' than it will show the DetailedView for a milisec and then after useEffect corrects the link this will show Overview component.

File: frontend/apps/web/src/components/system/Overview.tsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+                    ({ status, percentage }) => ({ name: status, percentage }),
+                  )}
+                />
+              ),
+            },
+            score: {
+              header: 'Score',
+              width: 150,
+              Cell: ({ data }) => (
+                <StyledButton
```
Comment (aleksaToljic) [inline frontend/apps/web/src/components/system/Overview.tsx:86]: Use getButton method so that we have that on one place.

File: frontend/packages/types/src/shared.ts [hunk 15]
```diff
@@ -16,6 +16,8 @@ export class PercentageSet {
 }
 
 export interface Percentage {
+  kind?: string
```
Comment (aleksaToljic) [inline frontend/packages/types/src/shared.ts:19]: Why is this added, Only model is using this data.
If we are using, than we should probably create another interface.

**Review by aleksaToljic — CHANGES_REQUESTED** [review]

**Comment by aleksaToljic** [comment]
I'll merge this for now, so that I have a preview of all the pages so far, and to fix what is needed.
